tools/csv-merger/csv-merger.go

```diff
@@ -910,6 +919,118 @@ func copyManifests() {
 	}
 }
 
+func getUXBackendServerDeployment() appsv1.DeploymentSpec {
+	replica := int32(1)
+	deployment := appsv1.DeploymentSpec{
+		Replicas: &replica,
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"app.kubernetes.io/component": "ux-backend-server",
+				"app.kubernetes.io/name":      "ux-backend-server",
+				"app":                         "ux-backend-server",
+			},
+		},
+		Strategy: appsv1.DeploymentStrategy{Type: appsv1.RecreateDeploymentStrategyType},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"app.kubernetes.io/component": "ux-backend-server",
+					"app.kubernetes.io/name":      "ux-backend-server",
+					"app.kubernetes.io/version":   "0.0.1",
+					"app":                         "ux-backend-server",
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name: "ux-backend-server",
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      "onboarding-private-key",
+								MountPath: "/etc/private-key",
+							},
+						},
+						Image:   *uxBackendServerImage,
+						Command: []string{"/usr/local/bin/ux-backend-server"},
+						Ports: []corev1.ContainerPort{
+							{
+								ContainerPort: 8080,
+							},
+						},
+						Env: []corev1.EnvVar{
+							{
+								Name:  "ONBOARDING_TOKEN_LIFETIME",
+								Value: os.Getenv("ONBOARDING_TOKEN_LIFETIME"),
+							},
+							{
+								Name:  "UX_BACKEND_PORT",
+								Value: os.Getenv("UX_BACKEND_PORT"),
+							},
+						},
+					},
+					{
+						Name: "oauth-proxy",
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      "ux-proxy-secret",
+								MountPath: "/etc/proxy/secrets",
+							},
+							{
+								Name:      "ux-cert-secret",
+								MountPath: "/etc/tls/private",
+							},
+						},
+						Image:           *uxBackendOauthImage,
+						ImagePullPolicy: "IfNotPresent",
+						Args: []string{"-provider=openshift",
+							"-https-address=:8888",
+							"-http-address=", "-email-domain=*",
+							"-upstream=http://localhost:8080/onboarding-tokens",
+							"-tls-cert=/etc/tls/private/tls.crt",
+							"-tls-key=/etc/tls/private/tls.key",
+							"-cookie-secret-file=/etc/proxy/secrets/session_secret",
+							"-openshift-service-account=ux-proxy-sa",
+							"-openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"},
+						Ports: []corev1.ContainerPort{
+							{
+								ContainerPort: 8888,
+							},
+						},
+					},
+				},
+				Volumes: []corev1.Volume{
+					{
+						Name: "onboarding-private-key",
+						VolumeSource: corev1.VolumeSource{
+							Secret: &corev1.SecretVolumeSource{
+								SecretName: "onboarding-private-key",
+							},
+						},
```

Comment:
if you are already getting this secret mounted by olm, what is the rbac for?

Reply:
ack, didn't know that, thanks.

Reply:
Not sure. I might be mistaken. We should test and then decide what to do.